### ISSUE TITLE
Less Massive Jumpsuit Pockets

### ISSUE THREAD

**Issue by bombasticSlacks** [CleverRaven/Cataclysm-DDA#53455]
### Is your feature request related to a problem? Please describe.

Currently a number of Normal Layer armors have massive amounts of storage. This isn't bad in and of itself but is a bit odd with trying to keep clothing layers consistent. There is a desire to make pockets under armor not be accessible or hard to access but most of the items with great pockets are high end crafted items so a first pass is just to make them more separated.

### Solution you would like.

I am currently working on this overhaul but wanted to put this together so I can reference back to it as I work on the piecemeal bits. There are a few separate projects:

1. Get it so that an armor can cover multiple layers. This is initially going to be for items that are on the strapped and outer layer but I'm going to write it in a way that it's general (maybe some day we can have less hard coded power armor).
2. Modify the offending armors in separate PR's each. **There is no intention to nerf anything just make it make more sense how things are distributed**
3. Finally make it so pockets that are blocked by higher layer items are _inaccessible_. So the move cost to use that pocket would be a portion of the cost to wear / remove all above items and then the actual pocket move cost as you dropped your exterior pants, unbuckled the plate that was covering it up, or started fishing around under the layers for the zipper. 

Progress so far:

- [x] Survivor https://github.com/CleverRaven/Cataclysm-DDA/pull/53471
- [x] Hub https://github.com/CleverRaven/Cataclysm-DDA/pull/53871
- [x] Scavenger https://github.com/CleverRaven/Cataclysm-DDA/pull/53533
- [x] Nomad

On that note the categories to be effected are:

#### Nomad Jumpsuits
| id              |
|-----------------|
| armor_nomad|
| armor_nomad_light|

The Nomad Jumpsuits are going to be getting most of their pockets removed and added to a new item called "Nomad Distributed Rig". The recipe for current jumpsuits will be split in two, with part of the materials, and time making the jumpsuit and part making the rig. Some cool new tools and goodies will be added to both the jumpsuit and rig making use of the EXODI focus on cyber power. The rig will be strapped and cover the upper arms, torso, thighs.

#### Survivor Jumpsuits
| id              |
|-----------------|
| lsurvivor_suit|
| xl_lsurvivor_suit|
| xlsurvivor_suit|
| wsurvivor_suit|
| xl_wsurvivor_suit|
| wsurvivor_suit_nofur|
| xl_wsurvivor_suit_nofur|
| h20survivor_suit|

The Survivor Jumpsuits are going to be getting most of their pockets removed and added to a new item called "Survivor Distributed Rig". The recipe for current jumpsuits will be split in two, with part of the materials, and time making the jumpsuit and part making the rig. The rig will be strapped and cover the lower torso and the thighs allowing for still wearing a survivor harness or load bearing vest with it.

#### Leather Armors
| id              |
|-----------------|
|armor_blarmor|
|armor_farmor|
|armor_larmor|
|armor_plarmor|

These are going to stay the same for now. The plated leather armor is a bit much but it will be effected by another set of changes I have lined up. 

#### Hub and Xedra Suits
| id              |
|-----------------|
|robofac_enviro_suit|
|activity_suit|

These suits if they should remain on the normal layer (which I think is fair), should have their storage separated out to the strapped layer for the most part. So same deal, explorers rig, will now be sold / drop where these currently do and be the bulk of their pockets on the strapped layer. 

#### Survivor Pants
| id              |
|-----------------|
|lsurvivor_pants|
|xl_lsurvivor_pants|
|pants_survivor|
|xl_pants_survivor|

~These have way too many pockets to be reckoned with and a leg only harness doesn't make much sense so I think the best solution for these is to make them like super pocketed kevlar ski pants and move them to the outer layer + strapped.~

had some feedback this should stay as is and I don't feel super strongly either way

#### Scavenger Armor
| id              |
|-----------------|
|armor_scavenger|

This armor is almost exclusively made out of items on the outer layer, I have no idea why it would be counted as normal layer in the first place. So I'm going to move it to outer+strapped and it can keep its pockets + its going to get ablative pockets equivalent to the US ballistic vest. 

#### Heavy Survivor Suit
This is going to get avoided and transformed when I'm doing my layering overhaul stuff based on thickness. I think it will be converted into heavily reinforced hard armor on the outer layer tho.

#### Of Note
Any of the items getting "Split" I plan to keep a deprecated old version around for current saves so there will be one old legacy item that is both parts and two new items that are the split. 

### Additional context

Again **I am currently working on this** just wanted all the info in one place so I can reference it in PR's

**Comment by catdach** [comment]
I don't agree with moving survivor pants to the outer+straped layer. I can't tell if you just mean to change the light and XL light versions of the pants or all 4 versions, but I don't think I like either being changed. The regular survivor pants AKA survivor cargo pants is constructed with a recipe that includes the regular "cargo/army pants"(which is a normal layer item btw) as well as multiple other storage items but ends up with **less** storage than the "cargo/army pants" almost the same volume but 2/3 the capacity for weight(16kg), in exchange for some pretty good protection and a bit more encumbrance. The light versions have 2/3 the volume, 1/2 the weight(8kg) and 3/4 the protection of the regular survivor pants to end up with the same encumbrance of regular cargo pants. And now it's going to conflict with items on the outer **and** strapped layers? I get how it seems like a lot of pockets for a _normal_ layer item, but those pockets are custom and really tiny. also I don't think it would be that much trouble to reach around a pair of kneepads and a holster to get to pockets on my pants.

If you just want to change the light and XL light versions (and I'm not sure why you would), then that just seems to make them less consistent with their non-light counterparts.

> **There is no intention to nerf anything just make it make more sense how things are distributed**

As much as you might not **intend** it, the layers of things is part of how they are balanced.

My understanding of layers(which could of course be completely wrong) is that it's supposed to scale in terms of (in general); the lower the layer the less encumbrance and greater the coverage, but the higher the layer the greater the encumbrance and greater the protection. Storage doesn't always fit into that dynamic very well, but in general high layer = more capacity and more encumbrance when full. Unless it's otherwise restricted by what can fit in it e.g. scabbards and bandoliers. Also higher layer items are going to get damaged more often, which makes storage on lower layers desirable.

**Comment by Photoloss** [comment]
> My understanding of layers(which could of course be completely wrong) is that it's supposed to scale in terms of (in general); the lower the layer the less encumbrance and greater the coverage, but the higher the layer the greater the encumbrance and greater the protection.

Not really, see e.g. bikinis and hazmat suits. Stronger armour is on the outer layers because it needs to be thick and bulky to be able to provide that protection, coverage is mostly just nerfed for balance reasons when a looted item serves as viable armour.

Pockets generally also get quite bulky fairly quickly, especially with the kinds of items we usually haul around. However I do think a lot of closer-layer items should have the ability to bear smaller pockets with stricter size limits. There's nothing really stopping a decently skilled survivor from sewing a dozen polo shirt pockets onto clothing and they wouldn't cause encumbrance while empty. Nor when filled with a bunch of ID cards, coins, seeds or the like but you'd struggle to fit a single flip phone, large pocket knife or vitamin bottle into any of them.

**Comment by bombasticSlacks** [comment]
I did mean to add all 4 recipes (I'm guessing the normal copy from the light) and I didn't realize they had lower storage than cargo pants. I don't feel strongly about the survivor pants I just searched for normal layer items with a lot of storage and they were on the list so I'll nix changing those. I get the argument that they are low profile in this case especially if they have similar volume to cargo pants.

> My understanding of layers(which could of course be completely wrong) is that it's supposed to scale in terms of (in general); the lower the layer the less encumbrance and greater the coverage, but the higher the layer the greater the encumbrance and greater the protection. Storage doesn't always fit into that dynamic very well, but in general high layer = more capacity and more encumbrance when full. Unless it's otherwise restricted by what can fit in it e.g. scabbards and bandoliers. Also higher layer items are going to get damaged more often, which makes storage on lower layers desirable.

This is how _balance_ is set up but the reason for this change is more logistic. If you are wearing full plate your normal layer pockets should be almost entirely inaccessible and beyond that you can't strap hard armor or a thick coat over big pockets comfortably which is the rule for layers not conflicting. If they aren't basically flush with the normal layer item it would cause discomfort. With that said keep in mind the next thing that is coming after this is pockets on items that are covered by other clothing are going to be harder / impossible to access. That's why for example the storage on the jumpsuits is being changed. I'm making the custom cool armors make sense before implementing the additional layers of simulation.

**Comment by catdach** [comment]
>This is how balance is set up but the reason for this change is more logistic. If you are wearing full plate your normal layer pockets should be almost entirely inaccessible and beyond that you can't strap hard armor or a thick coat over big pockets comfortably which is the rule for layers not conflicting. If they aren't basically flush with the normal layer item it would cause discomfort. With that said keep in mind the next thing that is coming after this is pockets on items that are covered by other clothing are going to be harder / impossible to access. That's why for example the storage on the jumpsuits is being changed. I'm making the custom cool armors make sense before implementing the additional layers of simulation.

Ok, that makes sense. My main gripe was the pants, everything else seems pretty reasonable. I do feel like there is a lack of late game gear to wear on the torso's normal layer, but that's a separate issue (and might just be me).

Keep up the good work. 👍

**Comment by bombasticSlacks** [comment]
alright I mean I did all the things, they are merged so I'm closing this
